src/main/java/teammates/ui/webapi/GetHasResponsesAction.java

```diff
@@ -55,11 +59,29 @@ void checkSpecificAccessControl() {
         } else {
             //An student can check whether he has submitted responses for a feedback session in his course.
             String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
-            String feedbackSessionName = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+            String feedbackSessionName = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
```

Comment:
There is too much nesting in `checkSpecificAccessControl` - let's clean this up a little to make it more readable. 

Do:
```
if (check) {

    return;
}

//rest of code
```

Instead of:
```
if (check) {

} else {
  //rest of code
}
```

Reply:
I will just remove any redundant `else` clause that is immediately succeeded by function return